apps/src/sharedFunctionalBlocks.js

```diff
@@ -320,19 +318,141 @@ function installString(blockly, generator) {
  * condition-value pairs before the default value.
  */
 function installCond(blockly, generator, numPairs) {
-  var blockName = 'functional_cond_' + numPairs;
+  // TODO(brent) - rtl
+
+  var blockName = 'functional_cond';
   blockly.Blocks[blockName] = {
     helpUrl: '',
     init: function() {
-      var args = [];
-      for (var i = 0; i < numPairs; i++) {
-        args.push({name: 'COND' + i, type: 'boolean', default: 'false'});
-        args.push({name: 'VALUE' + i, type: 'none', default: ''});
+      this.pairs_ = [];
+      this.setFunctional(true, {
+        headerHeight: 30
+      });
+
+      var options = {
+        fixedSize: { height: 35 }
+      };
+
+      var plusField = new Blockly.FieldIcon('+');
+      plusField.getRootElement().addEventListener('mousedown',
+        _.bind(this.addRow, this));
+
+      this.appendDummyInput()
+        .appendTitle(new Blockly.FieldLabel('cond', options))
+        .setAlign(Blockly.ALIGN_CENTRE);
+
+      this.appendFunctionalInput('DEFAULT');
+
+      this.appendDummyInput('PLUS')
+        .appendTitle(plusField)
+        .setInline(true);
+
+      this.setFunctionalOutput(true);
+
+      while(this.pairs_.length < numPairs) {
+        this.addRow();
+      }
+    },
+
+    /**
+     * Add another condition/value pair to the end.
+     */
+    addRow: function () {
+      // id is either the last value plus 1, or if we have no values yet 0
+      var id = this.pairs_.length > 0 ? (this.pairs_.slice(-1) * 1 + 1) : 0;
```

Comment:
What can id be other than this.pairs_.length?

Ah, or is this in the case they remove a condition in the middle of the block?